### PR TITLE
Adding Cypress tests for AMP metadata

### LINKED ISSUE #1375: Add Metadata Tests to AMP Pages
**Is your feature request related to a problem? Please describe.**
We currently don't test that we're serving the correct metadata on AMP pages. This will fix that.

**Describe the solution you'd like**
We will add tests into Cypress to make sure that our AMP pages have the relevant metadata.

**Testing notes**
Add Cypress tests to make sure the AMP pages are serving the relevant metadata.

Dev insight: Will Cypress tests be required or are unit tests sufficient? Will there be any potential regression? etc

### PR DESCRIPTION
**Pull request by jamesbrumpton**
Resolves #1375 

**Overall change:** _Adding Cypress tests for AMP metadata._

**Code changes:**

- _Adding Cypress tests, which check the metadata on AMP pages._

---

- [ ] I have assigned myself to this PR and the corresponding issues
- [ ] I have followed [the merging checklist](https://github.com/bbc/simorgh-infrastructure/blob/latest/documentation/MERGE_PROCESS.md) and this is ready to merge.

### CODE REVIEW

File: cypress/support/metaTestHelper.js [hunk 1]
```diff
@@ -94,18 +114,25 @@ export const metadataAssertion = () => {
...
-      lastPublished,
-    );
+    retrieveData(description, title, type, firstPublished, lastPublished);
     getElement('html').should('contain', language);
   });
 };
+
+export const retrieveAMPMetadata = data => {
+  const description = data.promo.summary;
+  // const { language } = data.metadata.passport;
```
Comment (dr3) [inline cypress/support/metaTestHelper.js]: ???

File: cypress/support/metaTestHelper.js [hunk 2]
```diff
@@ -94,18 +114,25 @@ export const metadataAssertion = () => {
...
-    retrieveMetaDataContent(
-      'head meta[name="article:modified_time"]',
-      lastPublished,
-    );
+    retrieveData(description, title, type, firstPublished, lastPublished);
     getElement('html').should('contain', language);
   });
 };
+
+export const retrieveAMPMetadata = data => {
```
Comment (dr3) [inline cypress/support/metaTestHelper.js]: Im not sure why metadataAssertion and retrieveAMPMetadata are seperate, they seem to do almost exactly the same things? Maybe they can be consolidated?

**Review by dr3 — CHANGES_REQUESTED** [review]

File: cypress/support/metaTestHelper.js [hunk 3]
```diff
@@ -81,31 +81,38 @@ export const retrieve404BodyResponse = (url, bodyResponse) => {
...
+    retrieveData(windowData);
     getElement('html').should('contain', language);
   });
 };
+
+export const metadataAssertionAMP = service => {
+  cy.window().then(win => {
+    const windowData = win.SIMORGH_DATA.data;
+    cy.visit(service);
+    retrieveData(windowData);
```
Comment (ChrisBAshton) [inline cypress/support/metaTestHelper.js]:
```suggestion
    checkDataMatchesMetadata(windowData);
```

**Review by ChrisBAshton — DISMISSED** [review]
I'd rename the function so it's clear what we're actually doing. But otherwise great! 👍

File: cypress/support/metaTestHelper.js [hunk 4]
```diff
@@ -81,31 +81,38 @@ export const retrieve404BodyResponse = (url, bodyResponse) => {
...
-    retrieveMetaDataContent(
-      'head meta[name="article:modified_time"]',
-      lastPublished,
-    );
+    retrieveData(windowData);
     getElement('html').should('contain', language);
   });
 };
+
+export const metadataAssertionAMP = service => {
```
Comment (sareh) [inline cypress/support/metaTestHelper.js]: We'll want to check the `lang` attribute exists on AMP pages as well as canonical pages, so I believe you can use just `metadataAssertion` and not need `metadataAssertionAMP`.

**Review by sareh — CHANGES_REQUESTED** [review]
Looks great so far. Just one comment added.

**Review by dr3 — APPROVED** [review]

**Review by sareh — APPROVED** [review]
👍 Looks great! Great work on investigating the differences in AMP and Canonical pages and then updating the Cypress tests to reflect that.

**Pull request merged**